File: backend/halfaday/src/main/java/com/xiaobantian/model/Place.java
package com.xiaobantian.model;

import jakarta.persistence.*;

@Entity
@Table(name = "places")
public class Place {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PlaceType type;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String address;
    private String phone;
    private String openingHours;
    private Integer avgPrice;
    private String imageUrls;
    private String mapUrl;
    private Double latitude;
    private Double longitude;

    public Place() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public PlaceType getType() { return type; }
    public void setType(PlaceType type) { this.type = type; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }

    public String getOpeningHours() { return openingHours; }
    public void setOpeningHours(String openingHours) { this.openingHours = openingHours; }

    public Integer getAvgPrice() { return avgPrice; }
    public void setAvgPrice(Integer avgPrice) { this.avgPrice = avgPrice; }

    public String getImageUrls() { return imageUrls; }
    public void setImageUrls(String imageUrls) { this.imageUrls = imageUrls; }

    public String getMapUrl() { return mapUrl; }
    public void setMapUrl(String mapUrl) { this.mapUrl = mapUrl; }

    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }

    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }
}

File: frontend/my-app/src/types/place.ts
export type PlaceType =
  | "SCENIC_SPOT"
  | "RESTAURANT"
  | "HOTEL"
  | "ACTIVITY";

export type PlaceUiType = "景點" | "餐飲" | "住宿" | "活動";

export interface Place {
  id: number;
  name: string;
  type: PlaceType;
  description: string | null;
  address: string | null;
  phone: string | null;
  openingHours: string | null;
  avgPrice: number | null;
  imageUrls: string | null;
  mapUrl: string | null;
  latitude: number | null;
  longitude: number | null;
  updatedAt?: string | null;
}

export interface CreatePlacePayload {
  name: string;
  type: PlaceType;
  description?: string | null;
  address?: string | null;
  phone?: string | null;
  openingHours?: string | null;
  avgPrice?: number | null;
  imageUrls?: string | null;
  mapUrl?: string | null;
  latitude?: number | null;
  longitude?: number | null;
}

export const PLACE_TYPE_LABEL: Record<PlaceType, PlaceUiType> = {
  SCENIC_SPOT: "景點",
  RESTAURANT: "餐飲",
  HOTEL: "住宿",
  ACTIVITY: "活動",
};

export const PLACE_TYPE_VALUE: Record<PlaceUiType, PlaceType> = {
  景點: "SCENIC_SPOT",
  餐飲: "RESTAURANT",
  住宿: "HOTEL",
  活動: "ACTIVITY",
};

export const PLACE_UI_TYPE_OPTIONS: Array<PlaceUiType | "全部"> = [
  "全部",
  "景點",
  "餐飲",
  "住宿",
  "活動",
];